src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSModelTypeNameConverter.java

```diff
@@ -0,0 +1,164 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.nodejs;
+
+import com.google.api.codegen.transformer.ModelTypeNameConverter;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypeNameConverter;
+import com.google.api.codegen.util.TypedValue;
+import com.google.api.codegen.util.nodejs.NodeJSTypeTable;
+import com.google.api.tools.framework.model.EnumValue;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
+
+public class NodeJSModelTypeNameConverter implements ModelTypeNameConverter {
+
+  /**
+   * A map from primitive types in proto to NodeJS counterparts.
+   */
+  private static final ImmutableMap<Type, String> PRIMITIVE_TYPE_MAP =
+      ImmutableMap.<Type, String>builder()
+          .put(Type.TYPE_BOOL, "boolean")
+          .put(Type.TYPE_DOUBLE, "number")
+          .put(Type.TYPE_FLOAT, "number")
+          .put(Type.TYPE_INT64, "number")
+          .put(Type.TYPE_UINT64, "number")
+          .put(Type.TYPE_SINT64, "number")
+          .put(Type.TYPE_FIXED64, "number")
+          .put(Type.TYPE_SFIXED64, "number")
+          .put(Type.TYPE_INT32, "number")
+          .put(Type.TYPE_UINT32, "number")
+          .put(Type.TYPE_SINT32, "number")
+          .put(Type.TYPE_FIXED32, "number")
+          .put(Type.TYPE_SFIXED32, "number")
+          .put(Type.TYPE_STRING, "String")
+          .put(Type.TYPE_BYTES, "String")
+          .build();
+
+  /**
+   * A map from primitive types in proto to zero value in NodeJS
+   */
+  private static final ImmutableMap<Type, String> PRIMITIVE_ZERO_VALUE =
+      ImmutableMap.<Type, String>builder()
+          .put(Type.TYPE_BOOL, "false")
+          .put(Type.TYPE_DOUBLE, "0.0")
+          .put(Type.TYPE_FLOAT, "0.0")
+          .put(Type.TYPE_INT64, "0")
+          .put(Type.TYPE_UINT64, "0")
+          .put(Type.TYPE_SINT64, "0")
+          .put(Type.TYPE_FIXED64, "0")
+          .put(Type.TYPE_SFIXED64, "0")
+          .put(Type.TYPE_INT32, "0")
+          .put(Type.TYPE_UINT32, "0")
+          .put(Type.TYPE_SINT32, "0")
+          .put(Type.TYPE_FIXED32, "0")
+          .put(Type.TYPE_SFIXED32, "0")
+          .put(Type.TYPE_STRING, "\'\'")
+          .put(Type.TYPE_BYTES, "\'\'")
+          .build();
+
+  private TypeNameConverter typeNameConverter;
```

Comment:
This is a copy from Php version. I think it serves a place holder where we can add more initialization here if needed in the future. Removing the constructor does not seem to bring obvious benefit, unless this style is a convention adopted by the team.